### PR TITLE
feat: add tron support with `send` and `approve` methods

### PR DESCRIPTION
**Pull request by Kozer4**
this PR will be continued by adding tests

### CODE REVIEW

File: src/bridge/index.ts [hunk 1]
```diff
@@ -1,29 +1,46 @@
...
 
-  async evmApprove(
-    web3: Web3,
+  async approve(
+    provider: Provider,
     approveData: ApproveData
   ): Promise<TransactionResponse> {
-    const evmBridge = new EvmBridge(this.api, web3);
-    return evmBridge.approve(approveData);
+    switch (provider.chainType) {
```
Comment (m-cote) [inline src/bridge/index.ts]: Consider if it's possible to replace switch with polymorphism.

File: src/bridge/trx/index.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    const swapAndBridgeMethod = bridgeContract.methods.swapAndBridge(
+      fromTokenAddress,
+      amount,
+      toAccountAddress,
+      toChainId,
+      toTokenAddress,
+      nonce,
+      messenger
+    );
+
```
Comment (m-cote) [inline src/bridge/trx/index.ts:57]: Do we need to estimate gas/energy here before sending transaction like we do it in the `EvmBridge` ?

File: src/bridge/trx/index.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  private getContract(contractAddress: string): Promise<any> {
+    return this.tronWeb.contract().at(contractAddress);
+  }
+
+  private async verifyTx(txId: string) {
+    const txReceipt = await this.getTransactionReceipt(txId);
+    if (txReceipt.receipt.result !== "SUCCESS") {
+      /* eslint-disable-next-line @typescript-eslint/restrict-template-expressions */
+      throw new Error(`Transaction status is ${txReceipt.receipt.result}`);
```
[the remark this answers is not in the document]
Comment (Kozer4) [reply src/bridge/trx/index.ts]: done

File: src/utils/calculation/index.ts [hunk 4]
```diff
@@ -2,6 +2,10 @@ import { Big, BigSource } from "big.js";
 import { TokenInfo } from "../../tokens-info";
 import { SYSTEM_PRECISION } from "./constants";
 
+export function sleep(ms: number): Promise<void> {
```
[the remark this answers is not in the document]
Comment (Kozer4) [reply src/utils/calculation/index.ts:5]: done

**Review by m-cote — APPROVED** [review]

**Pull request merged**